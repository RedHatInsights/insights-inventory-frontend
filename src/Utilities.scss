// Sass Imports

@import 'Utilities/_variables.scss';
@import 'Utilities/_colors.scss';
@import 'Utilities/_helpers.scss';
@import 'Utilities/_mixins.scss';

@import '~@redhat-cloud-services/frontend-components/index.css';
