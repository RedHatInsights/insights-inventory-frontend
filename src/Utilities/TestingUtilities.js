import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';

export const mountWithRouter = (Component, initialEntries) => {
  const wrapper = mount(
    <MemoryRouter initialEntriest={initialEntries}>{Component}</MemoryRouter>
  );

  return wrapper;
};

export const TestWrapper = ({
  children,
  routerProps = { initialEntries: ['/'] },
  path,
  store = {},
}) => {
  return (
    <MemoryRouter {...routerProps}>
      <Provider store={store}>
        {path ? (
          <Routes>
            <Route path={path} element={children} />
          </Routes>
        ) : (
          children
        )}
      </Provider>
    </MemoryRouter>
  );
};

TestWrapper.propTypes = {
  children: PropTypes.any.isRequired,
  routerProps: PropTypes.shape({
    initialEntries: PropTypes.arrayOf(PropTypes.string),
  }),
  path: PropTypes.string,
  store: PropTypes.any,
};
