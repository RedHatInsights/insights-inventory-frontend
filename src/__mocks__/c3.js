// eslint-disable-next-line no-undef
module.exports = () => 'c3';

