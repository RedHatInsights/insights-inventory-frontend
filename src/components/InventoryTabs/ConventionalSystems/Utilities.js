import flatMap from 'lodash/flatMap';
import {
  HOST_GROUP_CHIP,
  RHCD_FILTER_KEY,
  UPDATE_METHOD_KEY,
} from '../../../Utilities/constants';

const mapTags = ({ category, values }) =>
  values.map(
    ({ tagKey, value }) =>
      `${category ? `${category}/` : ''}${tagKey}${value ? `=${value}` : ''}`
  );

const filterMapper = {
  staleFilter: ({ staleFilter }, searchParams) =>
    staleFilter.forEach((item) => searchParams.append('status', item)),
  osFilter: ({ osFilter }, searchParams) => {
    // TODO This is very hackish. There should be a schema/feature on how the values translate into params
    const osParams = Object.entries(
      Object.values(osFilter).reduce((acc, item) => ({ ...acc, ...item }), {})
    )
      .filter(([, value]) => value === true)
      .map(([key]) => key);

    osParams.forEach((item) => {
      const keyParts = item.split('-');
      const [osName, value] = [
        keyParts.slice(0, keyParts.length - 2).join(' '),
        keyParts[keyParts.length - 1],
      ];

      if (value && osName) {
        searchParams.append('operating_system', `${osName}${value}`);
      }
    });
  },
  registeredWithFilter: ({ registeredWithFilter }, searchParams) =>
    registeredWithFilter?.forEach((item) =>
      searchParams.append('source', item)
    ),
  value: ({ value, filter }, searchParams) =>
    value === 'hostname_or_id' &&
    Boolean(filter) &&
    searchParams.append('hostname_or_id', filter),
  tagFilters: ({ tagFilters }, searchParams) =>
    tagFilters?.length > 0 &&
    searchParams.append('tags', flatMap(tagFilters, mapTags)),
  rhcdFilter: ({ rhcdFilter }, searchParams) =>
    rhcdFilter?.forEach((item) => searchParams.append(RHCD_FILTER_KEY, item)),
  lastSeenFilter: ({ lastSeenFilter }, searchParams) =>
    Object.keys(lastSeenFilter || {})?.forEach(
      (item) =>
        item === 'mark' &&
        searchParams.append('last_seen', lastSeenFilter[item])
    ),
  updateMethodFilter: ({ updateMethodFilter }, searchParams) =>
    updateMethodFilter?.forEach((item) =>
      searchParams.append(UPDATE_METHOD_KEY, item)
    ),
  hostGroupFilter: ({ hostGroupFilter }, searchParams) =>
    hostGroupFilter?.forEach((item) =>
      searchParams.append(HOST_GROUP_CHIP, item)
    ),
};

// TODO It might be better to not use a SearchParam instance and mutate it
// This should rather just return a plain object that can be
// applied with react routers useSearch params.
export const calculateFilters = (searchParams, filters = []) => {
  filters.forEach((filter) => {
    Object.keys(filter).forEach((key) => {
      filterMapper?.[key]?.(filter, searchParams);
    });
  });
  return searchParams;
};

export const calculatePagination = (searchParams, page, perPage) => {
  const currSearch = new URLSearchParams(location.search);
  const newPage = page !== undefined ? page : currSearch.get('page');
  const newPerPage =
    perPage !== undefined ? perPage : currSearch.get('per_page');
  !isNaN(parseInt(newPage)) && searchParams.append('page', newPage);
  !isNaN(parseInt(newPerPage)) && searchParams.append('per_page', newPerPage);
};
