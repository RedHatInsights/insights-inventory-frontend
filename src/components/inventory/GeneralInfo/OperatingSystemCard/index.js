export { default } from './OperatingSystemCard';
export * from './OperatingSystemCard';
