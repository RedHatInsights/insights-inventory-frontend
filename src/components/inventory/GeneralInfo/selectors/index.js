export { default } from './selectors';
export * from './selectors';
