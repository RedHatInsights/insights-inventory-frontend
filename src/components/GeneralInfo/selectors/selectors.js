/* eslint-disable camelcase */
import { registered } from '../../../Utilities/index';
import { verifyCollectorStaleness } from '../../../Utilities/sharedFunctions';
function safeParser(toParse, key) {
  try {
    return JSON.parse(toParse);
  } catch {
    return { [key]: toParse };
  }
}

export const propertiesSelector = (
  {
    number_of_cpus,
    number_of_sockets,
    cores_per_socket,
    ramSize,
    disk_devices,
    sap_sids,
    system_purpose,
    cpu_flags,
  } = {},
  { facts } = {}
) => ({
  cpuNumber: number_of_cpus || facts?.rhsm?.CPU_CORES,
  sockets: number_of_sockets || facts?.rhsm?.CPU_SOCKETS,
  coresPerSocket:
    cores_per_socket ||
    (facts?.rhsm?.CPU_CORES &&
      facts?.rhsm?.CPU_CORES &&
      Number(facts?.rhsm?.CPU_CORES, 10) /
        Number(facts?.rhsm?.CPU_SOCKETS, 10)),
  ramSize: ramSize || (facts?.rhsm?.MEMORY && `${facts?.rhsm?.MEMORY} GB`),
  storage:
    disk_devices &&
    disk_devices.map(({ device, label, mount_point, options, type }) => ({
      ...(device && safeParser(device, 'device')),
      label,
      ...(mount_point && safeParser(mount_point, 'mountpoint')),
      ...(options && safeParser(options, 'options')),
      ...(type && safeParser(type, 'mounttype')),
    })),
  sapIds: sap_sids,
  systemPurpose: system_purpose?.usage,
  cpuFlags: cpu_flags,
});

export const operatingSystem = (
  {
    arch,
    operating_system,
    os_kernel_version,
    last_boot_time,
    kernel_modules,
    system_update_method,
  } = {},
  { facts } = {}
) => ({
  release: operating_system,
  kernelRelease: os_kernel_version,
  architecture: arch || facts?.rhsm?.ARCHITECTURE,
  bootTime: last_boot_time,
  kernelModules: kernel_modules,
  systemUpdateMethod: system_update_method,
});

export const biosSelector = ({
  bios_vendor,
  bios_version,
  bios_release_date,
} = {}) => ({
  vendor: bios_vendor,
  version: bios_version,
  releaseDate: bios_release_date,
});

export const bootcSelector = ({ bootc_status } = {}) => ({
  bootedImage: bootc_status?.booted?.image,
  bootedImageDigest: bootc_status?.booted?.image_digest,
  stagedImage: bootc_status?.staged?.image,
  stagedImageDigest: bootc_status?.staged?.image_digest,
  rollbackImage: bootc_status?.rollback?.image,
  rollbackImageDigest: bootc_status?.rollback?.image_digest,
});

export const infrastructureSelector = (
  {
    infrastructure_type,
    infrastructure_vendor,
    public_ipv4_addresses,
    public_dns,
    network = {},
  } = {},
  { facts } = {}
) => ({
  type:
    infrastructure_type ||
    (facts?.rhsm?.IS_VIRTUAL !== undefined &&
      (facts?.rhsm?.IS_VIRTUAL ? 'virtual' : 'physical')) ||
    undefined,
  vendor: infrastructure_vendor,
  public_ipv4_addresses: public_ipv4_addresses,
  ipv4: network.ipv4,
  ipv6: network.ipv6,
  nics: network.interfaces,
  fqdn: public_dns,
});

export const configurationSelector = ({
  installed_packages,
  enabled_services,
  running_processes,
  repositories,
} = {}) => ({
  packages: installed_packages,
  services: enabled_services,
  processes: running_processes,
  repositories,
});

export const collectionInformationSelector = ({
  insights_client_version,
  insights_egg_version,
} = {}) => ({
  client: insights_client_version,
  egg: insights_egg_version,
});

export const getCollectorStatus = (collectorStaleness) => {
  return collectorStaleness
    ? verifyCollectorStaleness(collectorStaleness) !== 'Fresh'
      ? 'Stale'
      : 'Active'
    : 'N/A';
};

export const getDefaultCollectors = (entity) =>
  registered
    ?.filter((reporter) => reporter.label !== 'insights-client not connected')
    .map((reporter) => ({
      name: reporter.label,
      status: getCollectorStatus(
        entity?.per_reporter_staleness[reporter.value]
      ),
      updated: entity?.per_reporter_staleness[reporter.value]?.last_check_in,
      details: {
        name: reporter.idName,
        id: entity?.[reporter.idValue],
      },
    }));

export const systemStatus = ({ stale_timestamp } = {}) => ({
  stale: new Date() > new Date(stale_timestamp),
});
