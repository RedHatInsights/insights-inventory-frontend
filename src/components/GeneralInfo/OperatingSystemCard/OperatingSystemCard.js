import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoadingCard from '../LoadingCard';
import { generalMapper } from '../dataMapper';
import { operatingSystem } from '../selectors';
import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
import { extraShape, isDate } from '../../../constants';
import OperatingSystemFormatter from '../../../Utilities/OperatingSystemFormatter';

const OperatingSystemCardCore = ({
  detailLoaded,
  handleClick,
  hasRelease,
  hasKernelRelease,
  hasArchitecture,
  hasLastBoot,
  hasKernelModules,
  entity,
  extra,
  systemProfile,
}) => {
  const systemInfo = operatingSystem(systemProfile, entity);
  return (
    <LoadingCard
      title="Operating system"
      cardId="os-card"
      isLoading={!detailLoaded}
      items={[
        ...(hasRelease
          ? [
              {
                title: 'Release',
                value: (
                  <OperatingSystemFormatter
                    operatingSystem={systemInfo.release}
                  />
                ),
              },
            ]
          : []),
        ...(hasKernelRelease
          ? [{ title: 'Kernel release', value: systemInfo.kernelRelease }]
          : []),
        ...(hasArchitecture
          ? [{ title: 'Architecture', value: systemInfo.architecture }]
          : []),
        ...(hasLastBoot
          ? [
              {
                title: 'Last boot time',
                value: isDate(systemInfo.bootTime) ? (
                  <DateFormat date={systemInfo.bootTime} type="onlyDate" />
                ) : (
                  'Not available'
                ),
              },
            ]
          : []),
        ...(hasKernelModules
          ? [
              {
                title: 'Kernel modules',
                value: systemInfo.kernelModules?.length,
                singular: 'module',
                target: 'kernel_modules',
                onClick: () => {
                  handleClick(
                    'Kernel modules',
                    generalMapper(systemInfo.kernelModules, 'Module'),
                  );
                },
              },
            ]
          : []),
        ...(systemInfo.systemUpdateMethod
          ? [{ title: 'Update method', value: systemInfo.systemUpdateMethod }]
          : []),
        ...extra.map(({ onClick, ...item }) => ({
          ...item,
          ...(onClick && { onClick: (e) => onClick(e, handleClick) }),
        })),
      ]}
    />
  );
};

OperatingSystemCardCore.propTypes = {
  detailLoaded: PropTypes.bool,
  handleClick: PropTypes.func,
  hasRelease: PropTypes.bool,
  hasKernelRelease: PropTypes.bool,
  hasArchitecture: PropTypes.bool,
  hasLastBoot: PropTypes.bool,
  hasKernelModules: PropTypes.bool,
  extra: PropTypes.arrayOf(extraShape),
  entity: PropTypes.shape({
    facts: PropTypes.object,
  }),
  systemProfile: PropTypes.shape({
    arch: PropTypes.string,
    operating_system: PropTypes.string,
    os_kernel_version: PropTypes.string,
    last_boot_time: PropTypes.string,
    kernel_modules: PropTypes.arrayOf(PropTypes.string),
    system_update_method: PropTypes.string,
  }),
};
OperatingSystemCardCore.defaultProps = {
  detailLoaded: false,
  handleClick: () => undefined,
  hasRelease: true,
  hasKernelRelease: true,
  hasArchitecture: true,
  hasLastBoot: true,
  hasKernelModules: true,
  extra: [],
};

export const OperatingSystemCard = connect(
  ({ systemProfileStore: { systemProfile } }) => ({
    detailLoaded: systemProfile && systemProfile.loaded,
    systemProfile,
  }),
)(OperatingSystemCardCore);

OperatingSystemCard.propTypes = OperatingSystemCardCore.propTypes;
OperatingSystemCard.defaultProps = OperatingSystemCardCore.defaultProps;

export default OperatingSystemCard;
