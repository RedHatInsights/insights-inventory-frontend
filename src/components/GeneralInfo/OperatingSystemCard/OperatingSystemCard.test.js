import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import configureStore from 'redux-mock-store';
import { renderWithRouter } from '../../../Utilities/TestingUtilities';
import { osTest, rhsmFacts } from '../../../__mocks__/selectors';
import OperatingSystemCard from './OperatingSystemCard';

const location = {};

const fields = [
  'Release',
  'Kernel release',
  'Architecture',
  'Last boot time',
  'Kernel modules',
];

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => location,
  useParams: jest.fn(() => ({ modalId: 'kernel_modules' })),
}));

describe('OperatingSystemCard', () => {
  let initialState;
  let mockStore;

  beforeEach(() => {
    location.pathname = 'localhost:3000/example/path';
    mockStore = configureStore();
    initialState = {
      systemProfileStore: {
        systemProfile: {
          loaded: true,
          ...osTest,
        },
      },
      entityDetails: {
        entity: {
          facts: {
            rhsm: rhsmFacts,
          },
        },
      },
    };
  });

  it('should render correctly - no data', () => {
    const store = mockStore({ systemProfileStore: {}, entityDetails: {} });
    renderWithRouter(<OperatingSystemCard store={store} />);

    expect(
      screen
        .getAllByRole('definition')
        .reduce((prev, cur) => prev.concat(cur.textContent), '')
    ).toBe('');
  });

  it('should render correctly with data', () => {
    const store = mockStore(initialState);
    renderWithRouter(<OperatingSystemCard store={store} />);

    expect(
      screen
        .getAllByRole('definition')
        .reduce((prev, cur) => prev.concat(cur.textContent), '')
    ).toBe('test-releasetest-kerneltest-archNot available0 modules');
  });

  it('should render correctly with rhsm facts', () => {
    const store = mockStore({
      ...initialState,
      systemProfileStore: {
        systemProfile: {
          loaded: true,
        },
      },
    });

    renderWithRouter(<OperatingSystemCard store={store} />);
    expect(
      screen
        .getAllByRole('definition')
        .reduce((prev, cur) => prev.concat(cur.textContent), '')
    ).toBe('Not availableNot availablex86_64Not availableNot available');
  });

  describe('api', () => {
    it('should not render modules clickable', () => {
      const store = mockStore(initialState);
      renderWithRouter(<OperatingSystemCard store={store} />);

      expect(screen.queryByRole('link')).not.toBeInTheDocument();
    });

    it('should call handleClick on packages', async () => {
      initialState = {
        systemProfileStore: {
          systemProfile: {
            loaded: true,
            ...osTest,
            kernel_modules: ['some-module'],
          },
        },
        entityDetails: {
          entity: {
            facts: {
              rhsm: rhsmFacts,
            },
          },
        },
      };
      const store = mockStore(initialState);
      const onClick = jest.fn();
      location.pathname = 'localhost:3000/example/kernel_modules';
      renderWithRouter(
        <OperatingSystemCard handleClick={onClick} store={store} />
      );

      await userEvent.click(screen.getByRole('link'));
      expect(onClick).toHaveBeenCalled();
    });
  });

  [
    'hasRelease',
    'hasKernelRelease',
    'hasArchitecture',
    'hasLastBoot',
    'hasKernelModules',
  ].map((item, index) =>
    it(`should not render ${item}`, () => {
      const store = mockStore(initialState);
      renderWithRouter(
        <OperatingSystemCard store={store} {...{ [item]: false }} />
      );

      expect(
        screen.queryByRole('definition', { name: fields[index] })
      ).not.toBeInTheDocument();
    })
  );

  it('should render extra', () => {
    const store = mockStore(initialState);
    renderWithRouter(
      <OperatingSystemCard
        store={store}
        extra={[
          { title: 'something', value: 'test' },
          {
            title: 'with click',
            value: '1 tests',
            onClick: (_e, handleClick) => handleClick('Something', {}, 'small'),
          },
        ]}
      />
    );

    screen.getByRole('definition', {
      name: /something value/i,
    });
    screen.getByRole('link', {
      name: /1 tests/i,
    });
  });
});
