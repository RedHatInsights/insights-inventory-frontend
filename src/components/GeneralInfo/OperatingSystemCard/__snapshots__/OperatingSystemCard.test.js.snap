// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperatingSystemCard should not render hasArchitecture 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-9"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-9"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should not render hasKernelModules 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-11"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-11"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should not render hasKernelRelease 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-8"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-8"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should not render hasLastBoot 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-10"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-10"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should not render hasRelease 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-7"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-7"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should render correctly - no data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-1"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should render correctly with data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-2"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-2"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should render correctly with rhsm facts 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-4"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-4"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                Not available
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              x86_64
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should render enabled/disabled 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-3"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-3"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`OperatingSystemCard should render extra 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-12"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-12"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            Operating system
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Release
            </dt>
            <dd
              class=""
            >
              <span>
                test-release
              </span>
            </dd>
            <dt
              class=""
            >
              Kernel release
            </dt>
            <dd
              class=""
            >
              test-kernel
            </dd>
            <dt
              class=""
            >
              Architecture
            </dt>
            <dd
              class=""
            >
              test-arch
            </dd>
            <dt
              class=""
            >
              Last boot time
            </dt>
            <dd
              class=""
            >
              Not available
            </dd>
            <dt
              class=""
            >
              Kernel modules
            </dt>
            <dd
              class=""
            >
              0 modules
            </dd>
            <dt
              class=""
            >
              something
            </dt>
            <dd
              class=""
            >
              test
            </dd>
            <dt
              class=""
            >
              with click
            </dt>
            <dd
              class=""
            >
              <a
                href="http://localhost:5000//undefined"
              >
                1 tests
              </a>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;
